AppCenter/AppCenter/Internals/Context/AuthToken/MSAuthTokenContext.m

```diff
@@ -221,8 +228,12 @@ - (void)setAuthTokenHistory:(nullable NSArray<MSAuthTokenInfo *> *)authTokenHist
     self.authTokenHistoryArray = authTokenHistory;
     [MS_USER_DEFAULTS setObject:encryptedData forKey:kMSAuthTokenHistoryKey];
     MSLogDebug([MSAppCenter logTag], @"Saved new history state.");
+  } else if (decryptedData) {
```

Comment:
Isn't it `!decryptedData`?

Reply:
no. Line 218 is something extra and common to all branch in the nested `if` and `else`

Reply:
no. same as the previous function. we should not simply check null and log warning by each step. because people can pass in null.  we should log warning only when previous steps are not returning null but current step is returning null.

Reply:
but we will try to think it in a more intuitive way